common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -302,7 +307,7 @@ public V getNow() {
     @Override
     public boolean cancel(boolean mayInterruptIfRunning) {
         if (RESULT_UPDATER.get(this) == null &&
-                RESULT_UPDATER.compareAndSet(this, null, new CauseHolder(new CancellationException()))) {
+                RESULT_UPDATER.compareAndSet(this, null, new CauseHolder(CANCEL_EX))) {
```

Comment:
we can not make this a static instance as the user could access the exception and call addSuppressed(...) on it. This object would then never been garbage collected and so may cause OOME.